### PR TITLE
Issue 54: Add support for SA_KDF_ALGORITHM_COMMON_ROOT_KEY_LADDER

### PR DESCRIPTION
**Pull request by ericeberry**

### CODE REVIEW

**Review by riwoh — APPROVED** [review]

File: reference/src/client/test/sa_key_common.cpp [hunk 1]
```diff
@@ -61,6 +63,29 @@ bool SaKeyBase::get_root_key(std::vector<uint8_t>& key) {
...
+        }
+
+        common_root_key.resize(key_length);
+        status = true;
+    } else {
+        status = true;
+    }
+
+    key = common_root_key;
+    return status;
```
[the remark this answers is not in the document]
Comment (ericeberry) [reply reference/src/client/test/sa_key_common.cpp:86]: Done

File: reference/src/client/test/sa_key_common.cpp [hunk 2]
```diff
@@ -61,6 +63,25 @@ bool SaKeyBase::get_root_key(std::vector<uint8_t>& key) {
     return status;
 }
 
+bool SaKeyBase::get_common_root_key(std::vector<uint8_t>& key) {
+    if (common_root_key.empty()) {
+        char name[16];
```
Comment (mhabrat) [inline reference/src/client/test/sa_key_common.cpp:68]: Use enums for constant sizes.

**Review by mhabrat — CHANGES_REQUESTED** [review]
Can we move the root_keystore.p12 to a resource directory within the test folder?

File: reference/src/client/test/sa_key_common.cpp (hunk 2 above)
Comment (ericeberry) [reply reference/src/client/test/sa_key_common.cpp:68]: This is fixed in another PR.

**Comment by ericeberry** [comment]
keystore moved.

**Review by mhabrat — APPROVED** [review]

**Review by mhabrat — APPROVED** [review]

**Pull request merged**